src/main/java/com/google/sps/servlets/ClassServlet.java

```diff
@@ -0,0 +1,65 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.*; 
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreServiceConfig;
+
+
+/** Servlet that creates a new class Datastore.*/
+@WebServlet("/class")
+public class ClassServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      return;
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    // navigate to /_ah/admin to view Datastore
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    System.setProperty(DatastoreServiceConfig.DATASTORE_EMPTY_LIST_SUPPORT, Boolean.TRUE.toString());
+
+    String className = request.getParameter("className");
+    String enter = request.getParameter("enter");
+
+    if (enter.equals("create a new class!")){
+      ArrayList<String> emptyQueue = new ArrayList<String>();
+
+      Entity classEntity = new Entity("Class");
+      classEntity.setProperty("owner", "");
+      classEntity.setProperty("name", className);
+      classEntity.setProperty("beingHelped", "");
+      classEntity.setProperty("studentQueue", emptyQueue);
+      classEntity.setProperty("beingHelped", "");
+
+      datastore.put(classEntity);
+    } 
+    
+    else {
+      try {
+        Key classCode = KeyFactory.stringToKey(request.getParameter("classCode"));
```

Comment:
Would it be better to use `KeyFactory.createKey` here? Where is the entity type defined?

Reply:
Working under the assumption that the class code students use to enter the class will be the key associated with a given class (keys don't repeat between different types of entities).

Reply:
Ah I see. This could work, you just have to be careful how you return the key to the user.
If you use `entity.getKey().getId()` this will not work, this string does not encode the keys type. It will work if you use `KeyFactory.keyToString(entity.getKey())`. Either of these methods should work but we should choose one and be consistent across different tasks. We just need to clarify whether were giving users the whole key or just the id.

Reply:
We decided to use keys (instead of ID) to identify users/classes.